controllers/stargate/vector.go

```diff
@@ -0,0 +1,214 @@
+	ctx context.Context,
+	stargate api.Stargate,
+	actualDc *cassdcapi.CassandraDatacenter,
+	remoteClient client.Client,
+	dcLogger logr.Logger,
+) (ctrl.Result, error) {
+	namespace := stargate.Namespace
+	configMapKey := client.ObjectKey{
+		Namespace: namespace,
+		Name:      stargatepkg.VectorAgentConfigMapNameStargate(actualDc.Spec.ClusterName, actualDc.Name),
+	}
+	if stargate.Spec.Telemetry.IsVectorEnabled() {
+		// Create the vector toml config content
+		toml, err := CreateVectorToml(stargate.Spec.Telemetry)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+
+		desiredVectorConfigMap := stargatepkg.CreateVectorConfigMap(namespace, toml, *actualDc)
+		annotations.AddHashAnnotation(desiredVectorConfigMap)
+
+		// Check if the vector config map already exists
+		actualVectorConfigMap := &corev1.ConfigMap{}
+
+		if err := remoteClient.Get(ctx, configMapKey, actualVectorConfigMap); err != nil {
+			if errors.IsNotFound(err) {
+				if err := ctrl.SetControllerReference(&stargate, desiredVectorConfigMap, r.Scheme); err != nil {
+					dcLogger.Error(err, "Failed to set controller reference on new Stargate Vector ConfigMap", "ConfigMap", configMapKey)
+					return ctrl.Result{}, err
+				} else if err := remoteClient.Create(ctx, desiredVectorConfigMap); err != nil {
+					if errors.IsAlreadyExists(err) {
+						// the read from the local cache didn't catch that the resource was created
+						// already; simply requeue until the cache is up-to-date
+						dcLogger.Info("Vector Agent configuration already exists, requeueing")
+						return ctrl.Result{RequeueAfter: r.DefaultDelay}, nil
+					} else {
+						dcLogger.Error(err, "Failed to create Vector Agent ConfigMap")
+						return ctrl.Result{}, err
+					}
+				}
+				// Requeue to ensure the config map can be retrieved
+				return ctrl.Result{RequeueAfter: r.DefaultDelay}, nil
+			} else {
+				dcLogger.Error(err, "Failed to get Vector Agent ConfigMap")
+				return ctrl.Result{}, err
+			}
+		}
+
+		actualVectorConfigMap = actualVectorConfigMap.DeepCopy()
+
+		if !annotations.CompareHashAnnotations(actualVectorConfigMap, desiredVectorConfigMap) {
+			resourceVersion := actualVectorConfigMap.GetResourceVersion()
+			desiredVectorConfigMap.DeepCopyInto(actualVectorConfigMap)
+			actualVectorConfigMap.SetResourceVersion(resourceVersion)
+			if err := remoteClient.Update(ctx, actualVectorConfigMap); err != nil {
+				dcLogger.Error(err, "Failed to update Vector Agent ConfigMap resource")
+				return ctrl.Result{}, err
+			}
+			return ctrl.Result{RequeueAfter: r.DefaultDelay}, nil
+		}
+	} else {
+		if err := deleteConfigMapIfExists(ctx, remoteClient, configMapKey, dcLogger); err != nil {
+			return ctrl.Result{}, err
+		}
+	}
+
+	dcLogger.Info("Vector Agent ConfigMap successfully reconciled")
+	return ctrl.Result{}, nil
+}
+
+func deleteConfigMapIfExists(ctx context.Context, remoteClient client.Client, configMapKey client.ObjectKey, logger logr.Logger) error {
```

Comment:
We have a similar function already for K8ssandraCluster config maps, maybe we could extract this into a generic `deleteObjectIfExists`?

Reply:
Are you referring to the per_node config deletion here:
https://github.com/k8ssandra/k8ssandra-operator/blob/main/controllers/k8ssandra/per_node_config.go#L102-L111

Reply:
Hmm nevermind I think the function was deleted or refactored recently.